### PR TITLE
logging: parametrize workspace sku and resource specific tags

### PR DESCRIPTION
**Pull request by olljanat**
# Overview/Summary
I trying to get existing ALZ production environment to be managed with Bicep and do it on way that WhatIf functionality would result tell that code is in sync with environment.

This PR fixes some `Get-AzResourceGroupDeploymentWhatIfResult` results similarly than #190 on environment where workspace sku `pergb2018` is used and where automation account and analytics workspace have their own tags which are not used in elsewhere.

Related issues which are *not* fixed by this one:
* https://github.com/Azure/arm-template-whatif/issues/176
* https://github.com/Azure/arm-template-whatif/issues/251

## This PR fixes/adds/changes/removes

1. Add parameter `parLogAnalyticsWorkspaceSkuName` to logging module and default to `PerNode`
2. Add parameter `parAutomationAccountTags` to logging module and default to same value than `parTags`
3. Add parameter `parLogAnalyticsWorkspaceTags` to logging module and default to same value than `parTags`

### Breaking Changes
N/A

## Testing Evidence
**Before this PR:**
```powershell
Get-AzResourceGroupDeploymentWhatIfResult ...

The deployment will update the following scope:

Scope: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg-euw-shared-analytics

  ~ Microsoft.Automation/automationAccounts/aauto-euw-masterWorkspace-lalinked [2019-06-01]
    - tags.dataclassification: "confidential"
    x properties.sku.name:     "Basic"

  ~ Microsoft.OperationalInsights/workspaces/law-euw-masterWorkspace-476b2b [2020-08-01]
    - tags.sourceproject:     "Management"
    ~ properties.sku.name:    "pergb2018" => "PerNode"
```

**After this PR:**
Those changes was not shown.

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated tests *(if required)* [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml) - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows) - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Thanks @olljanat for raising.

Please see suggested/required changes. 

Look forward to seeing the updates and then reviewing again before merging 👍

Can you also update the parameter files please for this module?

**Comment by olljanat** [comment]
@jtracey93 thanks for good comments. Updated now.

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Thanks for the rapid turnaround @olljanat 👍

Could you update the parameters file please? https://github.com/Azure/ALZ-Bicep/blob/main/infra-as-code/bicep/modules/logging/logging.parameters.example.json

**Comment by olljanat** [comment]
@jtracey93 ah sorry, missed that comment. Now also parameters files are updated.

However I was wondering that is that actually considered as breaking change that now analytics workspace default sku will change from earlier hardcoded value `PerNode` to new default `PerGB2018` ?

**Comment by jtracey93** [comment]
> @jtracey93 Jack Tracey FTE ah sorry, missed that comment. Now also parameters files are updated.
> 
> However I was wondering that is that actually considered as breaking change that now analytics workspace default sku will change from earlier hardcoded value `PerNode` to new default `PerGB2018` ?

Thanks. Yup, this will be a breaking change 😔

File: infra-as-code/bicep/modules/logging/logging.parameters.example.json [hunk 1]
```diff
@@ -36,6 +39,16 @@
           "Environment": "POC"
       }
     },
+    "parAutomationAccountTags": {
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/logging/logging.parameters.example.json:42]: Let's emit/remove the `parAutomationAccountTags` and `parLogAnalyticsWorkspaceTags` and leave them to default

File: infra-as-code/bicep/modules/logging/logging.parameters.example.json [hunk 2]
```diff
@@ -36,6 +39,16 @@
...
+      "value": {
+          "Environment": "POC"
+      }
+    },
+    "parLogAnalyticsWorkspaceTags": {
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/logging/logging.parameters.example.json:47]: Let's emit/remove the `parAutomationAccountTags` and `parLogAnalyticsWorkspaceTags` and leave them to default

File: infra-as-code/bicep/modules/logging/mc-logging.parameters.example.json [hunk 3]
```diff
@@ -35,6 +38,16 @@
       "value": {
           "Environment": "POC"
       }
+    },
+    "parAutomationAccountTags": {
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/logging/mc-logging.parameters.example.json:42]: Let's emit/remove the `parAutomationAccountTags` and `parLogAnalyticsWorkspaceTags` and leave them to default

File: infra-as-code/bicep/modules/logging/mc-logging.parameters.example.json [hunk 4]
```diff
@@ -35,6 +38,16 @@
...
+      "value": {
+          "Environment": "POC"
+      }
+    },
+    "parLogAnalyticsWorkspaceTags": {
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/logging/mc-logging.parameters.example.json:47]: Let's emit/remove the `parAutomationAccountTags` and `parLogAnalyticsWorkspaceTags` and leave them to default

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Sorry @olljanat final change :)

File: infra-as-code/bicep/modules/logging/logging.parameters.example.json (hunk 1 above)
Comment (olljanat) [reply infra-as-code/bicep/modules/logging/logging.parameters.example.json:42]: Remove them totally from parameters file or to have `value: {}` on them?

Comment (jtracey93) [reply infra-as-code/bicep/modules/logging/logging.parameters.example.json:42]: remove them totally so they fall back to defaults in the module which is `parTags` 👍

Comment (olljanat) [reply infra-as-code/bicep/modules/logging/logging.parameters.example.json]: Done

**Review by jtracey93 — APPROVED** [review]
LGTM 🚢

**Pull request merged**